lib/stages/RfiFrameDrop.hpp

```diff
@@ -90,15 +90,16 @@ class RfiFrameDrop : public kotekan::Stage {
     /// Output buffer to receive baseline subset visibilities
     Buffer* _buf_out;
 
-    /// Thresholds
-    struct ThresholdData {
-        std::vector<std::tuple<float, size_t, float>> thresholds;
-        std::vector<size_t> sk_exceeds;
-    };
-    std::shared_ptr<ThresholdData> _thresholds;
+    /// Everything that gets changed by dynamic config updates,
+    /// including the resulting dataset state ID, is protected by a mutex.
+    /// Thresholds and fractions are kept in the dataset state directly to avoid duplicates.
+    std::mutex update_mutex;
+    std::vector<size_t> num_sk;
+    std::vector<size_t> sk_exceeds;
+    state_id_t state_id;
+    const RFIFrameDropState* state_ptr;
```

Comment:
So you keep the entire state of the thresholds in here now? Well I guess along side `num_sk`.

Reply:
Do we provide any lifetime guarantees on bare pointers that come out of the dataset manager? I'm wondering if there's theoretically any issue here that could allow us to read an invalid pointer.

I mean, I don't think there's any issue at the moment, but if this is a pattern that we think should be fine, so we need to make a guarantee in the dataset manager that the target will never be free'd.

Reply:
Yes, `num_sk` and `sk_exceeds` are not part of the state,a s we discussed.

Reply:
This variable never gets initialized.

Reply:
@andrerenard I think it is. `RfiFrameDrop::rest_enable_callback` is implicitly called when the callbacks are bound at the end of the constructor. @nritsche should check though.

Reply:
That could work, but neither of those call backs initialize it as far as I can tell.  And since it's `const *` it has to be set in the constructor.   Also it segfaults when I run it.

Reply:
Ah, I see now. It's initialised in one of the callbacks (`rest_thresholds_callback`) here ( https://github.com/kotekan/kotekan/blob/a57de492b013714d26669cca2b92474f63f3209d/lib/stages/RfiFrameDrop.cpp#L333), but it's actually used in the other one (`rest_enable_callback`) first.

Reply:
Oh yeah. And that one even tries to initialise using data from the previous state.